packages/plugin-ext/src/plugin/telemetry-ext.ts

```diff
@@ -0,0 +1,373 @@
+            let updatedProperty = value.replace(/%20/g, ' ');
+
+            // First we anonymize any possible file paths
+            updatedProperty = anonymizeFilePaths(updatedProperty, cleanUpPatterns);
+
+            // Then we do a simple regex replace with the defined patterns
+            for (const regexp of cleanUpPatterns) {
+                updatedProperty = updatedProperty.replace(regexp, '');
+            }
+
+            // Lastly, remove commonly leaked PII
+            updatedProperty = removePropertiesWithPossibleUserInfo(updatedProperty);
+
+            return updatedProperty;
+        }
+        return undefined;
+    }, new Set());
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function cloneAndChange(obj: any, changer: (orig: any) => any, seen: Set<any>): any {
+    // impossible to clone a null or undefined object
+    if (isUndefined(obj) || obj === null) {
+        return obj;
+    }
+
+    const changed = changer(obj);
+    if (!isUndefined(changed)) {
+        return changed;
+    }
+
+    if (Array.isArray(obj)) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const r1: any[] = [];
+        for (const e of obj) {
+            r1.push(cloneAndChange(e, changer, seen));
+        }
+        return r1;
+    }
+
+    if (isObject(obj)) {
+        if (seen.has(obj)) {
+            throw new Error('Cannot clone recursive data-structure');
+        }
+        seen.add(obj);
+        const r2 = {};
+        for (const i2 in obj) {
+            if (Object.prototype.hasOwnProperty.call(obj, i2)) {
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                (r2 as any)[i2] = cloneAndChange(obj[i2], changer, seen);
+            }
+        }
+        seen.delete(obj);
+        return r2;
+    }
+
+    return obj;
+}
+
+/**
+ * Copies all properties of source into destination. The optional parameter "overwrite" allows to control
+ * if existing properties on the destination should be overwritten or not. Defaults to true (overwrite).
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function mixin(destination: any, source: any, overwrite: boolean = true): any {
+    if (!isObject(destination)) {
+        return source;
+    }
+
+    if (isObject(source)) {
+        Object.keys(source).forEach(key => {
+            if (key in destination) {
+                if (overwrite) {
+                    if (isObject(destination[key]) && isObject(source[key])) {
+                        mixin(destination[key], source[key], overwrite);
+                    } else {
+                        destination[key] = source[key];
+                    }
+                }
+            } else {
+                destination[key] = source[key];
+            }
+        });
+    }
+    return destination;
+}
+
+type UnknownObject<T extends object> = Record<string | number | symbol, unknown> & { [K in keyof T]: unknown };
```

Comment:
Duplicate code from:

https://github.com/eclipse-theia/theia/blob/a0e45ce63ed240af512efdcf58bd772835b7b65c/packages/core/src/common/types.ts#L20

Reply:
No need for this type anymore, as this was only used by a removed function (isObject). It was removed.